app/controllers/api/v1/move_events_controller.rb

```diff
@@ -51,27 +58,34 @@ def events
         else
           render status: :bad_request,
                  json: {
-                   errors: [{ title: 'invalid event_name', detail: "#{event_name} is not supported" }],
+                   errors: [{ title: 'invalid event_name', detail: 'event_name is not supported' }],
                  }
         end
       end
 
     private
 
-      def validate_params
+      def validate_params!(event_params, require_from_location: false, require_to_location: false)
         MoveEvents::ParamsValidator.new(event_params).validate!
+        params.require(:data).require(:relationships).require(:from_location).require(:data).require(:id) if require_from_location
+        params.require(:data).require(:relationships).require(:to_location).require(:data).require(:id) if require_to_location
+      end
+
+      def complete_params
+        @complete_params ||= params.require(:data).permit(COMPLETE_PARAMS).to_h
       end
 
-      def event_params
-        @event_params ||= params.require(:data).permit(PERMITTED_EVENT_PARAMS).to_h
+      def lockout_params
+        @lockout_params ||= params.require(:data).permit(LOCKOUT_PARAMS).to_h
       end
 
-      def event_name
-        @event_name ||= event_params.dig(:attributes, :event_name)
+      def redirect_params
+        @redirect_params ||= params.require(:data).permit(REDIRECT_PARAMS).to_h
       end
 
-      def timestamp
-        @timestamp ||= Time.zone.parse(event_params.dig(:attributes, :timestamp))
+      def depreciated_event_params
+        # TODO: deleteme once FE updated
+        @depreciated_event_params ||= params.require(:data).permit(DEPRECIATED_EVENT_PARAMS).to_h
```

Comment:
`deprecated` 😁

Reply:
oops, schoolboy error - oh well, they will be deleted soon